pkg/cmd/admin/top/imagestreams.go

```diff
@@ -0,0 +1,167 @@
+package top
+
+import (
+	"fmt"
+	"io"
+
+	gonum "github.com/gonum/graph"
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/api/graph"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+)
+
+const (
+	TopImageStreamsRecommendedName = "imagestreams"
+
+	topImageStreamsLong = `Show usage statistics for ImageStreams
+
+This command analyzes all the ImageStreams managed by the platform and presents current
+usage statistics.`
+
+	topImageStreamsExample = `  # Show usage statistics for ImageStreams
+  %[1]s %[2]s`
+)
+
+// NewCmdTopImageStreams implements the OpenShift cli top imagestreams command.
+func NewCmdTopImageStreams(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	opts := &TopImageStreamsOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   "Show usage statistics for ImageStreams",
+		Long:    topImageStreamsLong,
+		Example: fmt.Sprintf(topImageStreamsExample, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(opts.Complete(f, cmd, args, out))
+			kcmdutil.CheckErr(opts.Validate(cmd))
+			kcmdutil.CheckErr(opts.Run())
+		},
+	}
+
+	return cmd
+}
+
+type TopImageStreamsOptions struct {
+	// internal values
+	Images  *imageapi.ImageList
+	Streams *imageapi.ImageStreamList
+
+	// helpers
+	out      io.Writer
+	osClient client.Interface
+}
+
+// Complete turns a partially defined TopImageStreamsOptions into a solvent structure
+// which can be validated and used for showing limits usage.
+func (o *TopImageStreamsOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	namespace := cmd.Flag("namespace").Value.String()
+	if len(namespace) == 0 {
+		namespace = kapi.NamespaceAll
+	}
+	o.out = out
+
+	allImages, err := osClient.Images().List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Images = allImages
+
+	allStreams, err := osClient.ImageStreams(namespace).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Streams = allStreams
+
+	return nil
+}
+
+// Validate ensures that a TopImageStreamsOptions is valid and can be used to execute command.
+func (o TopImageStreamsOptions) Validate(cmd *cobra.Command) error {
+	return nil
+}
+
+// Run contains all the necessary functionality to show current image references.
+func (o TopImageStreamsOptions) Run() error {
+	infos := o.imageStreamsTop()
+	Print(o.out, ImageStreamColumns, infos)
+	return nil
+}
+
+var ImageStreamColumns = []string{"NAME", "STORAGE", "IMAGES", "LAYERS"}
```

Comment:
So with limit range `openshift/ImageStream: Max[openshift.io/image-tags]=20,
Max[openshift.io/images]=50` set on namespace `joe`, it could look like this:

```
NAME               TAG USAGE   IMAGE USAGE    STORAGE     IMAGES  LAYERS
openshift/python                              1.21GiB     4       36
joe/ruby           1/20        3/50           717.76MiB   3       27
```